h2o-py/h2o/estimators/generic.py

```diff
@@ -24,8 +24,8 @@ def __init__(self, **kwargs):
         super(H2OGenericEstimator, self).__init__()
         self._parms = {}
         names_list = {"model_id", "model_key", "path"}
-        if(all(name is None for name in names_list)):
-                raise H2OValueError("At least one of model-generic")
+        if(all(kwargs.get(name, None) is None for name in [ "model_key", "path"])):
+                raise H2OValueError("At least one of [\"model_key\", \"path\"] is required.")
```

Comment:
btw ideally, parameters validation should go in the `train` method because user can usually create an estimator with empty constructor and set the params using the setters (this is actually the standard way when used in a sklearn pipeline), but we don't have any hook today to ensure that validation is done in the right place: I'll probably enforce this in my ongoing PR.